### ISSUE TITLE
[BUG] Call of the Wild does not reset when boss resets

### ISSUE THREAD

**Issue by ToxicKevinFerm** [JamminL/cata-classic-bugs#1268]
#### 1.1. Description
Call of the Wild is a pet ability. It has a 5 minute CD, and it should reset when a boss resets.
This is also related to #1242 which was fixed recently.
  #### 1.2. How to Reproduce
1. Pull a boss
2. Use Call of the Wild
3. Wipe/reset after 30 seconds
  #### 1.3  Source Material
It's easy to test
#### 2.1. Description
It should reset when other cooldowns does. 
It actually used to at some point in Wrath, but after a patch it regressed.
  #### 2.2 Source Material
Don't have any sadly

**Comment by Kenniflopper** [comment]
Would appreciate some news regarding https://github.com/JamminL/cata-classic-bugs/issues/1172 aswell.

**Comment by MannySJH** [comment]
Have you tested this recently? I raided yesterday on prepatch and was pretty sure call of the wild was resetting.
https://classic.warcraftlogs.com/reports/LQkWFYDdfqNv24zj#fight=1&type=damage-done&source=8
It was available on LDW after using it on marrowgar when it shouldn't have been usually.

**Comment by AidanZMoon** [comment]
So it never got regressed in Wrath. It is working how it always has since cooldown resets. People just noticed it in Naxx always being available, and then started wiping in Ulduar/ToGC/ICC because Naxx was super easy and the others had hard mode and progression.

Call of the Wild will not reset unless your pet is alive when Encounter_End fires.

So when you kill a boss it will reset.
If you die and your pet despawns, and your raid wipes after that your pet is not considered present and it's as if it "released" so the effect cannot hit something that is not there.

The reason your pet has sated cleared even though it was dead, is because it checks the master for sated and if no sated on master, wipe your own.

So this is not a bug, it's limited by how encounters end and a pet not being present when you wipe, it dies, or you die.

This is why you'll see reports of others saying it's fine.

**Comment by ToxicKevinFerm** [comment]
> So this is not a bug, it's limited by how encounters end and a pet not being present when you wipe, it dies, or you die.


This makes sense. 
I would still suggest that it's something that should be changed. Since the pet is not despawned (dead on the ground) when the Encounter_End event is fired, is the unit not available for selection as players are, or is it totally impossible?

**Comment by AidanZMoon** [comment]
Let me have a look for you Kevin. I THINK that is how it works if the pet's body is still present and you didn't dismiss it, die yourself, or try to revive it and canceled the revive. But let me check since I'm not gonna trust memory on this and give a definitive "that's already how it works."

**Comment by ToxicKevinFerm** [comment]
Thank you!

I think I understand the logic of when it happens or doesn't happen now.
At least in Cataclysm it's easier to run with several pets we can swap around, so I suppose it's not as big of a deal. It's a bit unfortunate that the most common outcome of a wipe, you dying with the pet still alive wouldn't work, but we can work around it.

**Comment by Kenniflopper** [comment]
> Thank you!
> 
> I think I understand the logic of when it happens or doesn't happen now. At least in Cataclysm it's easier to run with several pets we can swap around, so I suppose it's not as big of a deal. It's a bit unfortunate that the most common outcome of a wipe, you dying with the pet still alive wouldn't work, but we can work around it.

Currently you cant swap around pets cause they all share cooldowns, so you cant use call of the wild on a different pet cus it will share cooldown with ur first pet. i reported this weeks ago, but no news yet about it.

**Comment by AidanZMoon** [comment]
Yeah just tested it and can give you the situations it will reset Call of the Wild:
**WILL RESET**
- You end the encounter alive with your pet alive.
- You end the encounter alive with your pet dead and the body still present. (You must not attempt to and fail to finish a cast of Revive Pet or else the body disappears during the cast time).
-----> This includes your pet dying and you being Feign Deathed. That will reset the pet's cooldown as long as the body is present.
- Your pet dies, THEN you die. (The pet's body will still be there as opposed to if you die with your pet still alive the body is dismissed).
- The boss is leashed or resets.
- You swap pets to a different pet and then the encounter_ends (this one is the craziest because your pet is not active but it will have its cooldown reset. So it looks like it just has to hit ANY of your pets to reset the cooldowns across all of them since they are sharing cooldowns.)

**WILL NOT RESET**
- You die before your pet, which causes it to be dismissed.
- You dismiss your pet during the fight and end the fight without it active.
- Your pet dies, you take a Battle Res, and the fight ends. (When you take a battle res with a dead pet, your pet's body is despawned).
- You die, take a Battle Res, and do not call your pet.
- You swap pets, you die, the encounter_ends. (Even though that pet did not die, you have no active pet for spells to be reset on).


TL;DR
You get the cooldown back if the encounter_end hits any of your pet's bodies. Doesn't matter if alive or dead, but it MUST be present in some way shape or form. The most common ways it is not present are:
You die with it still alive and it is dismissed while your dead body is on the floor during encounter_end.
It dies, then you die and take a Battle Res and don't have time to cast a full Revive Pet.

**Comment by Kisanen** [comment]
> * So it looks like it just has to hit ANY of your pets to reset the cooldowns across all of them since they are sharing cooldowns.)

That's a bug, In Cataclysm, pets should not share cooldowns; they should have their individual cooldowns. Could you please check this in your reference client?

**Comment by ToxicKevinFerm** [comment]
Just a thought then, since the CD is shared, you don't require the pet itself to reset the CD, instead could you reset it for all pets on Encounter_End?

**Comment by MannySJH** [comment]
> You die before your pet, which causes it to be dismissed.
You dismiss your pet during the fight and end the fight without it active.

Are there any plans to address this? The first point you mentioned is going to be the most likely scenario during progression by far and not having a substantial pet cd ready each pull is going to be a huge hit to hunters and even the raid in general.

**Comment by AidanZMoon** [comment]
> Just a thought then, since the CD is shared, you don't require the pet itself to reset the CD, instead could you reset it for all pets on Encounter_End?

That is how it works.
If you have any pet active, all of your pets reset their cooldowns on encounter end, but you must have a pet alive or their body present.

**Comment by ToxicKevinFerm** [comment]
> > Just a thought then, since the CD is shared, you don't require the pet itself to reset the CD, instead could you reset it for all pets on Encounter_End?
> 
> That is how it works. If you have any pet active, all of your pets reset their cooldowns on encounter end, but you must have a pet's body present (alive or dead).

That's how it works today, but in most cases this condition will never be true, meaning for the purpose of progress, the pet never resets it's CDs realistically. 

I meant that since the pets share CDs, you shouldn't require a pet to be present at all, since for all intents and purposes, it doesn't matter, since they share a CD. So you could reset the CD for all pets, wether you have one or not.

We can close this thread and I'll open a new one with #somechanges

**Comment by ToxicKevinFerm** [comment]
Sorry i forgot to close it
